sockeye/beam_search.py

```diff
@@ -677,7 +679,12 @@ def forward(self,
             target_dists, model_states, target_factors = self._inference.decode_step(best_word_indices,
                                                                                      model_states,
                                                                                      vocab_slice_ids)
-
+            # Do not generate <unk> token if prevent_unk is specified
+            if self.prevent_unk:
+                if target_dists.ndim == 1:
+                    target_dists[C.UNK_ID] = float('inf')
```

Comment:
in what cases would `ndim` be 1?

Reply:
when `beam_size =1` and `batch_size = 1`. The [unit test for beam_search ](https://github.com/awslabs/sockeye/blob/master/test/unit/test_beam_search.py#L307)use `beam_size=1` and `batch_size =1` so I add this case to make sure it passes the test.

Reply:
tested with `float('inf')` and `np.inf`, and also with using an `adding_mask` on GPU. Updated the code with adding mask since it's faster.